core/controllers/learner_dashboard_test.py

```diff
@@ -301,10 +300,36 @@ def test_get_message_summaries(self):
         messages_summary = response_dict['message_summary_list']
 
         # Check the summary of the second message.
+        self.assertEqual(len(messages_summary), 2)
         second_message = messages_summary[1]
         self.assertDictContainsSubset({
             'text': 'Message 1',
             'author_username': 'editor'
         }, second_message)
 
         self.logout()
+
+    def test_anonymous_feedback_is_recorded_correctly(self):
+        self.post_json(
+            '/explorehandler/give_feedback/%s' % self.EXP_ID_1,
+            {
+                'state_name': '',
```

Comment:
Do we need state_name for feedback? (I think payload should include "include_author", right?) Also, is the test coverage is 100% now? If yes then you can change the PR title else I think you can work on making it 100%.

Reply:
Actually what happens if you don't give "state_name"? I guess it shouldn't complain, but if it does.... that might be a different issue which would need investigation 😅. Could you confirm the behaviour @varun-tandon?

Reply:
I'm fairly certain you do not need to provide a state name. This has not caused any issues, and since this is for testing, I'm pretty sure it isn't relevant to the test.

Reply:
I think the state name is optional, but I'm surprised that '' is accepted. Does the frontend actually pass in '' or None/null in that case?

The test suggests that we aren't doing enough controller-level validation which might be something that's worth fixing. I.e. add a check in the controller to reject invalid values for the state_name property.

Reply:
I didn't find any validation for the payload values in the backend and even didn't find any "state_name" payload. 
![image](https://user-images.githubusercontent.com/16653571/52324895-0ea0c400-2a09-11e9-945d-cb6c1631cb34.png)

Reply:
Oh, ok. So why are we passing a state name?

Reply:
I pulled this code from another similar test I believe (I cannot remember which one) so I was unsure if state name was optional. I am happy to remove it if the consensus is that it’s confusing for this test to include it.

Reply:
Done